python/lsst/ip/isr/isrFunctions.py

```diff
@@ -546,10 +560,30 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
         stdevBiasArr = 0.74*(percentiles[2] - percentiles[0])  # robust stdev
         diff = numpy.abs(biasArray - medianBiasArr[:, numpy.newaxis])
         biasMaskedArr = numpy.ma.masked_where(diff > numSigmaClip*stdevBiasArr[:, numpy.newaxis], biasArray)
-        collapsed = numpy.mean(biasMaskedArr, axis=1)
-        if collapsed.mask.sum() > 0:
-            collapsed.data[collapsed.mask] = numpy.mean(biasArray.data[collapsed.mask], axis=1)
+
+        if fitType in ('MEDIAN_PER_ROW',):
+            collapsed = []
+            fitTypeStats = afwMath.stringToStatisticsProperty('MEDIAN')
+            # Loop over rows because there is no afwMath.Median(array, axis=1)
+            for row in biasMaskedArr:
+                if row.mask is False:
```

Comment:
I need to go in with a debugger, but is that _ever_ `True`? Isn't `row.mask` the mask array for a whole row? If so, that never `is` False.

Reply:
Actually, one of the unit tests found that it could be `False` if this condition is not satisfied: https://github.com/lsst/ip_isr/blob/tickets/DM-23238/python/lsst/ip/isr/isrFunctions.py#L545 (a masked numpy array is not created).

Reply:
It might `== False` but surely it can't _be_ `False`, right?

Reply:
_i.e._ remember that `0 == False` is True, but `0 is False` is not.

Reply:
You have to be careful if a mask is in theory a numpy array here. There are many definitions of "truth" available depending on whether there is any numpy mask attached and whether it's filled with zeroes. If mask is meant to be a numpy array I would not expect it to ever be truly the boolean `False`.  Is this really testing that `mask` is `None` rather than `False`?